app/src/main/java/com/davidtomas/taskyapp/features/auth/domain/useCase/ValidateInputsUseCase.kt

```diff
@@ -0,0 +1,55 @@
+package com.davidtomas.taskyapp.features.auth.domain.useCase
+
+import com.davidtomas.taskyapp.core.domain.model.Result
+import com.davidtomas.taskyapp.features.auth.domain.model.InputValidationError
+import com.davidtomas.taskyapp.features.auth.domain.model.InputValidationErrors
+import com.davidtomas.taskyapp.features.auth.domain.model.InputValidationType
+
+class ValidateInputsUseCase(
```

Comment:
I'd maybe prefer a bit less generic name, since just from reading this class you don't know which inputs you mean. How about `ValidateRegistrationFieldsUseCase`?